report/chap/parity.tex

```diff
@@ -1,95 +0,0 @@
-    Given a (possibly history dependent) winning strategy $\rho$ for \Player{i} in $(\G^f_p,v)$, there is a finite memory winning strategy $\tilde{\rho}$ for \Player{i} in $(\G_p,v)$.
-\end{theorem}
-
-The following stack based technique will be used to construct the $\tilde{\rho}$. This will be useful later too. Notice that $\rho$ is only defined (or relevant) for simple paths starting in $v$.
-
-\subsection{Stack based extension} 
-\label{sec:stack}
-Let $\rho$ be a strategy for \Player{i} defined on simple paths. The following describes $ \tilde{\rho}$ -- an extension of $\rho$ to arbitrary plays using finite memory. At each stage \Player{i} maintains a simple path of $G$ in its memory. If the game starts at $v_0$, let $m_0=v_0$. At the \nth[k+1]\ stage, if it's \Player{i}'s turn, play $v_{k+1}=\rho(m_k)$ (otherwise the opponent decides $v_{k+1}$), and update the memory as follows.\\
-Assume
-\begin{align*}
-    m_k &=u_0 u_1 \ldots u_r \quad r \geq 0\\
-    \intertext{then}
-    m_{k+1}&=\begin{cases}
-        u_0 u_1 \ldots u_s & \text{ if } u_s = v_{k+1} \text{ for some } s \leq r \\
-        u_0  u_1 \ldots u_r v_{k+1} & \text{ otherwise }
-        \end{cases}
-\end{align*}
-In the first case $u_{s+1} \ldots u_r v_{k+1}$ forms a cycle in $G$, which had to be eliminated for $m_{k+1}$ to remain a simple path.
-
-For each $n$, $m_n$ is a simple path from $v_0$ to $v_n$ which conforms with $\rho$. If $C_1,C_2 \ldots C_{r_n}$ are the cycles eliminated by the \nth\ round then $m_n, C_1, C_2 \ldots C_r$ forms a partition of $v_0v_1 \ldots v_n$
-
-
-\begin{proof}[Proof of Theorem \ref{thm:parityDet}]
-    Obtain $\tilde{\rho}$ from $\rho$ by the above stack based extension. We will show that this strategy $ \tilde{\rho}$ is winning for \Player{i} in $(\G_p,v)$.
-
-    Assume $\rho$ is a strategy for \Player{0}. The proof for \Player{1} is similar. Let $\pi = v v_1 v_2 \ldots $ be an infinite path that conforms with $\tilde{\rho}$. We will show that \mip\ for $\pi$ is even. This shows that $\tilde{\rho}$ is winning for \Player{0} in $(\G_p,v)$.
-    
-    Since $\rho$ is winning for \Player{0} in $(\G^f_p,v)$, for any path that starts at $v$ and conforms with $\rho$ till the first cycle formed, the largest priority of the cycle will be even. In the stack based implementation of $ \tilde{\rho}$ on $\pi$, each $m_i$ conforms with $\rho$, hence the max priority in each of the eliminated cycles $C_r$ will be even. Given this, the \mip\ of $\pi$ cannot be odd. Indeed, let $u$ be a vertex which is visited infinitely often in $\pi$, and has same priority as \mip\ of $\pi$. Then $u$ must be a part of infinitely many $C_i$'s. But if $p(u)$ is odd, then there will be a vertex in each of these cycles which has priority strictly greater than $p(u)$ (since max priority in every $C_i$ is even). Since $C_i$'s correspond to disjoint positions of $\pi$, this shows that the \mip\ of $\pi$ is strictly greater than $p(u)$. A contradiction.
-
-\end{proof}
```

Comment:
For proofs/results which have been rewritten since the original version was
published, you can use the reference from which your proof is inspired (if
this is not relevant just refer to any one which has a proof).

For reference to formalisms (like parity games/mean-payoff games), you
should as far as possible try to cite the paper which introduced it.

On Wed, Jun 8, 2016 at 9:19 PM, Miheer Dewaskar <notifications@github.com>
wrote:

> In report/chap/parity.tex
> <https://github.com/miheerdew/master-thesis/pull/2#discussion_r66282554>:
>
> > -        u_0  u_1 \ldots u_r v_{k+1} & \text{ otherwise }
> > -        \end{cases}
> > -\end{align*}
> > -In the first case $u_{s+1} \ldots u_r v_{k+1}$ forms a cycle in $G$, which had to be eliminated for $m_{k+1}$ to remain a simple path.
> > -
> > -For each $n$, $m_n$ is a simple path from $v_0$ to $v_n$ which conforms with $\rho$. If $C_1,C_2 \ldots C_{r_n}$ are the cycles eliminated by the \nth\ round then $m_n, C_1, C_2 \ldots C_r$ forms a partition of $v_0v_1 \ldots v_n$
> > -
> > -
> > -\begin{proof}[Proof of Theorem \ref{thm:parityDet}]
> > -    Obtain $\tilde{\rho}$ from $\rho$ by the above stack based extension. We will show that this strategy $ \tilde{\rho}$ is winning for \Player{i} in $(\G_p,v)$.
> > -
> > -    Assume $\rho$ is a strategy for \Player{0}. The proof for \Player{1} is similar. Let $\pi = v v_1 v_2 \ldots $ be an infinite path that conforms with $\tilde{\rho}$. We will show that \mip\ for $\pi$ is even. This shows that $\tilde{\rho}$ is winning for \Player{0} in $(\G_p,v)$.
> > -
> > -    Since $\rho$ is winning for \Player{0} in $(\G^f_p,v)$, for any path that starts at $v$ and conforms with $\rho$ till the first cycle formed, the largest priority of the cycle will be even. In the stack based implementation of $ \tilde{\rho}$ on $\pi$, each $m_i$ conforms with $\rho$, hence the max priority in each of the eliminated cycles $C_r$ will be even. Given this, the \mip\ of $\pi$ cannot be odd. Indeed, let $u$ be a vertex which is visited infinitely often in $\pi$, and has same priority as \mip\ of $\pi$. Then $u$ must be a part of infinitely many $C_i$'s. But if $p(u)$ is odd, then there will be a vertex in each of these cycles which has priority strictly greater than $p(u)$ (since max priority in every $C_i$ is even). Since $C_i$'s correspond to disjoint positions of $\pi$, this shows that the \mip\ of $\pi$ is strictly greater than $p(u)$. A contradiction.
> > -
> > -\end{proof}
>
> Can the references be any references or should they be the original ones?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/miheerdew/master-thesis/pull/2/files/b878539b6ce25a0a09d9f3fb9ddfc86c3eae60d1#r66282554>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS346BuQji8ofpIL4_hLzJOZoZzgP-KMks5qJuSVgaJpZM4IuINp>
> .
>